### PR TITLE
feat(vscode-ext): intellisense for design tokens

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Related issues

- :rocket: Feature #1420

### CODE REVIEW

File: apps/vscode-extension/src/intellisense/design-token.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { CompletionItem, CompletionItemKind, CompletionItemProvider, HoverProvider } from 'vscode';
+import { DesignTokenVariableSet, DesignTokenVariableStructure, getCssTokenValueRenderer, parseDesignTokenFile } from '@o3r/design';
+import * as vscode from 'vscode';
+import * as path from 'node:path';
+
+const readFile = async (filePat: string) => (await vscode.workspace.fs.readFile(vscode.Uri.parse(filePat))).toString();
+
+const getDesignTokens = async (currentFile?: string) => {
```
Comment (kpanot) [inline apps/vscode-extension/src/intellisense/design-token.ts:8]: I think you should add a cache mechanism for the token calculation for the case we will have a large amount of files.
A simple way it to recalculate the Design Token only if one of the file of the glob has changed

File: apps/vscode-extension/package.json [hunk 2]
```diff
@@ -21,6 +21,13 @@
     "configuration": {
       "title": "Otter",
       "properties": {
+        "otter.design.filesPatterns": {
+          "type": "array",
+          "default": [
+            "**/*.token.json"
```
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply apps/vscode-extension/package.json:27]: Could be done in another task I think

File: apps/vscode-extension/src/intellisense/design-token.ts (hunk 1 above)
Comment (matthieu-crouzet) [reply apps/vscode-extension/src/intellisense/design-token.ts:8]: Will be done in another [task](https://github.com/AmadeusITGroup/otter/issues/1490)

**Review by kpanot — APPROVED** [review]

**Review by vscaiceanu-1a — APPROVED** [review]
Documentation can be improved via https://github.com/AmadeusITGroup/otter/issues/1505

**Pull request merged**